code/game/objects/explosion.dm

```diff
@@ -215,3 +199,73 @@ var/explosion_shake_message_cooldown = 0
 
 	explosion_destroy_multi_z(epicenter, offcenter, devastation_range / 2, heavy_impact_range / 2, light_impact_range / 2, flash_range / 2, explosion_time)
 	explosion_destroy_multi_z(epicenter, offcenter, devastation_range / 2, heavy_impact_range / 2, light_impact_range / 2, flash_range / 2, explosion_time, whodunnit)
+
+/proc/CalculateExplosionBlock(list/affected_turfs)
+	. = list()
+	// we cache the explosion block rating of every turf in the explosion area
+	//explosion block reduces explosion distance based on path from epicentre
+	for(var/affected_turf in 1 to length(affected_turfs))
```

Comment:
```suggestion
	for(var/turf/affected_turf in affected_turfs)
```
Why not do this? Is it slower?

Reply:
That is slower, but `for(var/turf/affected_turf as anything in affected_turfs)` would be even faster than what's used here